crates/bevy_ecs/src/system/mod.rs

```diff
@@ -241,6 +241,42 @@ mod tests {
         system.initialize(&mut world);
     }
 
+    #[test]
+    #[should_panic]
+    fn crazy_dance7() {
+        let mut world = World::new();
+        let sys = |_um: &World, mut _no: &mut World| {
+            // this should not compile
```

Comment:
Tests that shouldn't compile will block CI, rather than panicking :) You need bevy_ecs_compile_fail_tests for this instead.

Reply:
ah, it should just panic